exist-core/src/main/java/org/exist/xquery/functions/map/SingleKeyMapType.java

```diff
@@ -59,12 +60,16 @@ public int size() {
     }
 
     @Override
-    public AbstractMapType remove(AtomicValue key) {
-        try {
+    public AbstractMapType remove(AtomicValue[] keysAtomicValues) throws XPathException {
+        for (final AtomicValue key: keysAtomicValues) {
```

Comment:
I do not understand the logic here :-)

Reply:
For some reason Maps with just one key have their own java representation.
`let $a := map { 'a': 1 }` 
this implementation ensures that `map:remove($a, ('z','zz','zzz'))`, `map:remove($a, ('z','a'))` and `map:remove($a, 'a')` behave as expected.